pkg/logql/downstream.go

```diff
@@ -584,6 +583,37 @@ func ResultStepEvaluator(res logqlmodel.Result, params Params) (StepEvaluator, e
 	}
 }
 
+// ProbabilisticResultStepEvaluator coerces a downstream StepResult into a
+// ProbabilisticStepEvaluator
+func ProbabilisticResultStepEvaluator(res logqlmodel.Result, params Params) (ProbabilisticStepEvaluator, error) {
+	var (
+		start = params.Start()
+	)
+
+	switch data := res.Data.(type) {
+	case promql.Vector:
+		var exhausted bool
+		return NewProbabilisticStepEvaluator(func() (bool, int64, StepResult) {
+			if !exhausted {
+				exhausted = true
+				return true, start.UnixNano() / int64(time.Millisecond), SampleVector(data)
+			}
+			return false, 0, nil
+		}, nil, nil)
```

Comment:
how come we're supporting promql vector?

Reply:
This is future proof instead we are sending other queries to the probabilistic endpoint. I'm happy to remove this again.